Makefile

```diff
@@ -146,3 +147,59 @@ release-skip-tests quick-release:
 	KUBE_RELEASE_RUN_TESTS=n KUBE_FASTBUILD=true build/release.sh
 .PHONY: release-skip-tests quick-release
 
+
+
+################# HACKING BELOW HERE ###########################
+#FIXME: once we have multiple kinds of generation, this can probably become
+#       API_DIRS or similar
+#FIXME: could fill this from a `find` that looks for a //+k8s:gen_deep_copy tag
+DEEP_COPY_DIRS := \
+	pkg/api \
+	pkg/api/v1 \
+	pkg/apis/authentication.k8s.io \
+	pkg/apis/authentication.k8s.io/v1beta1 \
+	pkg/apis/authorization \
+	pkg/apis/authorization/v1beta1 \
+	pkg/apis/autoscaling \
+	pkg/apis/autoscaling/v1 \
+	pkg/apis/batch \
+	pkg/apis/batch/v1 \
+	pkg/apis/batch/v2alpha1 \
+	pkg/apis/apps \
+	pkg/apis/apps/v1alpha1 \
+	pkg/apis/componentconfig \
+	pkg/apis/componentconfig/v1alpha1 \
+	pkg/apis/policy \
+	pkg/apis/policy/v1alpha1 \
+	pkg/apis/extensions \
+	pkg/apis/extensions/v1beta1 \
+	pkg/apis/metrics \
+	pkg/apis/metrics/v1alpha1 \
+	pkg/apis/rbac \
+	pkg/apis/rbac/v1alpha1 \
+	federation/apis/federation \
+	federation/apis/federation/v1alpha1
+
+PRJ_SRC_PATH := k8s.io/kubernetes
+GENERATED_FILE_PREFIX := zz_generated.
+DEEP_COPY_BASENAME := $(GENERATED_FILE_PREFIX)deep_copy
+DEEP_COPY_FILENAME := $(DEEP_COPY_BASENAME).go
+
+#FIXME: document why we always build this, or else test deps, build bin, remake deps
+gen_deep_copy:
+	@make -s WHAT=cmd/libs/go2idl/deepcopy-gen
+	@make -s $(addsuffix /$(DEEP_COPY_FILENAME), $(DEEP_COPY_DIRS))
+
+# For each dir in DEEP_COPY_DIRS, this generates a rule of the form:
+#     path/to/dir/$(DEEP_COPY_FILENAME): <all files in dir, except generated ones>
+# This has to be done in two steps because wildcards don't seem to work in
+# static pattern rules.
+# FIXME: the grep is to avoid circular dep.  Could autogen into a different
```

Comment:
@wojtek-t @lavalamp the args part of go2idl says that output pkg is supported, but it looks like nothing but set-gen supports it.  Trivial attempts at changing the output path for deepcopy did not work.  Clue me in?  Or shoot this idea down.  It would be nice to have the generated files in a wholly separate pkg I think.